File: mpi-hello/mpi_hello.c
#include <mpi.h>
#include <stdio.h>

int main (int argc, char *argv[]) {

  int ierr, nranks, myid;

  ierr = MPI_Init(&argc, &argv);
  ierr = MPI_Comm_size(MPI_COMM_WORLD, &nranks);
  ierr = MPI_Comm_rank(MPI_COMM_WORLD, &myid);
  printf("Hello from rank %i of %i\n", \
         myid, nranks);
  ierr = MPI_Finalize();

  return 0;
}

File: f90-mpi-hello/mpi_hello.f90
PROGRAM mpi_hello

  USE mpi
  IMPLICIT NONE

  ! Internal variables
  INTEGER :: nranks, myid, ierr

  CALL MPI_INIT( ierr )

  CALL MPI_COMM_SIZE( MPI_COMM_WORLD, nranks, ierr )
  CALL MPI_COMM_RANK( MPI_COMM_WORLD, myid, ierr )

  PRINT *, 'Hello from rank ', myid, ' of ', nranks

  CALL MPI_FINALIZE( ierr )

  STOP
END PROGRAM mpi_hello
